src/main/java/fi/helsinki/cs/tmc/core/communication/ExerciseSubmitter.java

```diff
@@ -108,43 +112,53 @@ public String submit(String currentPath)
      * Submits folder of exercise to TMC. Finds it from current directory.
      *
      * @param currentPath path from which this was called.
-     * @param observer {@link ProgressObserver} that is informed of the submission progress
-     * @return String with url from which to get results or null if exercise was not found.
+     * @param observer {@link ProgressObserver} that is informed of the
+     * submission progress
+     * @return URL from which to get results or null if exercise was
+     * not found.
      * @throws IOException if zip creation fails
```

Comment:
I'd say, for public api (there is like one class we consider to be public api in core) we should tell about unchecked exceptions we throw. Otherwise for trivial cases I'd exclude those from javadoc.